cmake_modules/BuildUtils.cmake

```diff
@@ -0,0 +1,212 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Borrowed the file from Apache Arrow:
+# https://github.com/apache/arrow/blob/main/cpp/cmake_modules/BuildUtils.cmake
+
+function(iceberg_install_cmake_package PACKAGE_NAME EXPORT_NAME)
```

Comment:
These are boilerplate code to add libraries with essential arguments for includes, linking libraries, and exporting install targets. I copied them from Apache Arrow with simplification. They are super useful if we will add more libraries in the future. In the next step, I will add third-party libraries and these lines will be used so I don't want to delete them for now and add them back in the next patch :)

Reply:
I think @raulcd is familiar with these lines :)

Reply:
> Not much to add here, as @wgtmac pointed out those are utility functions to help adding new libraries in the future. The ADD_ICEBERG_LIB function is a highly configurable function that simplifies the process of defining, building, and installing both shared and static libraries, along with their associated settings (e.g., include directories, link dependencies, etc.). To be honest if we only plan to build iceberg core and puffin at the moment it might be overkill but I don't think it's harmful to reuse what Arrow has done here.

I also think it is overkill, I suggest we stick to modern cmake(which is target oriented) and start from easy, improve as project gets complicate.

Reply:
Isn't it target-oriented already?

Reply:
> Isn't it target-oriented already?

I guess not, you are using *Set* for adding source files, there is a target_sources which might be a better option. I don't insist if you think it's ok. In general, it LGTM.

Reply:
Thanks for the suggestion! I have switched to use `target_sources`.